FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -87,20 +87,13 @@ final class MessageService {
                 isEmail: true
             )
 
-            let isWrongPassPhraseError = decrypted.blocks.first(where: { block -> Bool in
-                guard let errorBlock = block.decryptErr, case .needPassphrase = errorBlock.error.type else {
-                    return false
-                }
-                return true
-            })
-
-            if isWrongPassPhraseError != nil {
-                reject(MessageServiceError.wrongPassPhrase(rawMimeData, passPhrase))
-            } else {
-                self.savePassPhrases(value: passPhrase, with: keys)
-                let processedMessage = try self.processMessage(rawMimeData: rawMimeData, with: decrypted, keys: keys)
-                resolve(processedMessage)
+            guard !self.hasWrongPassPhraseError(decrypted) else {
+                return reject(MessageServiceError.wrongPassPhrase(rawMimeData, passPhrase))
             }
+
+            self.savePassPhrases(value: passPhrase, with: keys)
```

Comment:
Pass phrase should be saved at the point I'm entering it - as long as it can unlock a particular key (keys), then it should be saved for that key (keys) - that happens before decryption is even attempted. 

The result of decryption of particular message or attachment should not affect saving of pass phrase in any way.

Reply:
https://github.com/FlowCrypt/flowcrypt-ios/pull/640/commits/194c176a50d81ad2e4cb66897fef29e1dbb3f42f so passphrases should be saved before message decrypt, right?